pkg/provider/securitygroups.go

```diff
@@ -100,7 +106,7 @@ func (p *SecurityGroupProvider) getSecurityGroups(ctx context.Context, filters [
 		return nil, fmt.Errorf("describing security groups %+v, %w", filters, err)
 	}
 	p.cache.SetDefault(fmt.Sprint(hash), output.SecurityGroups)
-	if p.cm.HasChanged("security-groups", output.SecurityGroups) {
+	if p.cm.HasChanged("security-groups (provisioner-%s)", output.SecurityGroups) {
```

Comment:
This seems odd to me that we'd append it with provisioner, considering it's only defined within the Node Template. Is this defined elsewhere?